src/algebra/order/with_zero.lean

```diff
@@ -178,6 +178,16 @@ instance : linear_ordered_add_comm_group_with_top (additive (order_dual α)) :=
   ..additive.linear_ordered_add_comm_monoid_with_top,
   ..additive.nontrivial }
 
+instance linear_ordered_comm_group_with_zero.to_covariant_class : covariant_class α α (*) (≤) :=
+{ elim := λ a b c bc, linear_ordered_comm_group_with_zero.mul_le_mul_left _ _ bc a }
+
+instance {β : Type*} [linear_ordered_comm_group β] :
+  linear_ordered_comm_group_with_zero (with_zero β) :=
+{ mul_le_mul_left := λ x y, mul_le_mul_left',
+  zero_le_one     := with_zero.zero_le _,
+  ..with_zero.linear_order,
+  ..with_zero.comm_group_with_zero }
```

Comment:
There's a missing `monoid` version of this instance I think too:
```suggestion
instance {β : Type*} [linear_ordered_comm_monoid β] :
  linear_ordered_comm_monoid_with_zero (with_zero β) :=
{ mul_le_mul_left := λ x y, mul_le_mul_left',
  zero_le_one     := with_zero.zero_le _,
  ..with_zero.linear_order,
  ..with_zero.comm_monoid_with_zero }

instance {β : Type*} [linear_ordered_comm_group β] :
  linear_ordered_comm_group_with_zero (with_zero β) :=
{ ..with_zero.linear_ordered_comm_monoid_with_zero,
  ..with_zero.comm_group_with_zero }
```

Also I think these might be better all the way up on line 54, where you can use `α` instead of introducing `β`

Reply:
Thanks! I have moved them up as suggested.